### ISSUE TITLE
CocoaPods fails to validate SnapshotTesting 1.8.0+

### ISSUE THREAD

**Issue by stephencelis** [pointfreeco/swift-snapshot-testing#350]
We just released 1.8.0, which includes support for SwiftUI. Unfortunately, we can't push to CocoaPods trunk due to validation errors:

```
 -> SnapshotTesting (1.8.0)
    - ERROR | [iOS] xcodebuild: Returned an unsuccessful exit code.
    - NOTE  | xcodebuild:  note: Using new build system
    - NOTE  | xcodebuild:  note: Building targets in parallel
    - NOTE  | xcodebuild:  note: Planning build
    - NOTE  | xcodebuild:  note: Constructing build description
    - ERROR | [iOS] xcodebuild:  SnapshotTesting/Sources/SnapshotTesting/Snapshotting/SwiftUIView.swift:19:45: error: no type named 'View' in module 'SwiftUI'
    - ERROR | [iOS] xcodebuild:  SnapshotTesting/Sources/SnapshotTesting/Snapshotting/SceneKit.swift:35:12: error: ambiguous use of 'image'
    - NOTE  | xcodebuild:  SnapshotTesting/Sources/SnapshotTesting/Snapshotting/SwiftUIView.swift:33:22: note: found this candidate
    - NOTE  | xcodebuild:  SnapshotTesting/Sources/SnapshotTesting/Snapshotting/UIView.swift:17:22: note: found this candidate
    - ERROR | [iOS] xcodebuild:  SnapshotTesting/Sources/SnapshotTesting/Snapshotting/SpriteKit.swift:35:12: error: ambiguous use of 'image'
    - ERROR | [iOS] xcodebuild:  SnapshotTesting/Sources/SnapshotTesting/Snapshotting/SwiftUIView.swift:60:24: error: use of unresolved identifier 'UIHostingController'
    - ERROR | [iOS] xcodebuild:  SnapshotTesting/Sources/SnapshotTesting/Snapshotting/SwiftUIView.swift:64:35: error: use of unresolved identifier 'UIHostingController'
    - ERROR | [iOS] xcodebuild:  SnapshotTesting/Sources/SnapshotTesting/Snapshotting/UIView.swift:7:13: error: ambiguous use of 'image'
    - ERROR | [iOS] xcodebuild:  SnapshotTesting/Sources/SnapshotTesting/Snapshotting/UIViewController.swift:7:13: error: ambiguous use of 'image'
    - NOTE  | [iOS] xcodebuild:  SnapshotTesting/Sources/SnapshotTesting/Snapshotting/UIViewController.swift:43:22: note: found this candidate
    - NOTE  | xcodebuild:  warning: Skipping code signing because the target does not have an Info.plist file and one is not being generated automatically. (in target 'App' from project 'App')
    - NOTE  | xcodebuild:  note: Execution policy exception registration failed and was skipped: Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted" (in target 'SnapshotTesting' from project 'Pods')
    - NOTE  | xcodebuild:  note: Execution policy exception registration failed and was skipped: Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted" (in target 'Pods-App' from project 'Pods')
    - NOTE  | xcodebuild:  note: Execution policy exception registration failed and was skipped: Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted" (in target 'App' from project 'App')

[!] The spec did not pass validation, due to 8 errors.
```

It seems related to the inclusion of SwiftUI code, but we guard this with `#if canImport(SwiftUI)`. Even more puzzling, after running `pod lib lint --no-clean --verbose` and re-running the `xcodebuild` command that fails, I get a successful build and return code. We'd love to continue to serve our CocoaPods users, but need a lil help here.

**Comment by thedavidharris** [comment]
It actually breaks Carthage views as well. Conditional linking of SwiftUI has always been really weird in frameworks.

```
/Carthage/Checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Snapshotting/SwiftUIView.swift:19:45: error: no type named 'View' in module 'SwiftUI'
extension Snapshotting where Value: SwiftUI.View, Format == UIImage {
                                            ^
/Carthage/Checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Snapshotting/SceneKit.swift:35:12: error: ambiguous use of 'image'
    return Snapshotting<View, Image>.image(precision: precision).pullback { scene in
           ^
/Carthage/Checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Snapshotting/SwiftUIView.swift:33:22: note: found this candidate
  public static func image(
                     ^
/Carthage/Checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Snapshotting/UIView.swift:17:22: note: found this candidate
  public static func image(
                     ^
/Users/dharr419/workspace/mobileapp-ios/Carthage/Checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Snapshotting/SpriteKit.swift:35:12: error: ambiguous use of 'image'
    return Snapshotting<View, Image>.image(precision: precision).pullback { scene in
           ^
/Carthage/Checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Snapshotting/SwiftUIView.swift:33:22: note: found this candidate
  public static func image(
                     ^
/Carthage/Checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Snapshotting/UIView.swift:17:22: note: found this candidate
  public static func image(
                     ^
/Carthage/Checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Snapshotting/SwiftUIView.swift:60:24: error: use of unresolved identifier 'UIHostingController'
          controller = UIHostingController.init(
                       ^~~~~~~~~~~~~~~~~~~
/Carthage/Checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Snapshotting/SwiftUIView.swift:64:35: error: use of unresolved identifier 'UIHostingController'
          let hostingController = UIHostingController.init(rootView: view)
                                  ^~~~~~~~~~~~~~~~~~~
/Carthage/Checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Snapshotting/UIView.swift:7:13: error: ambiguous use of 'image'
    return .image()
            ^
/Carthage/Checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Snapshotting/SwiftUIView.swift:33:22: note: found this candidate
  public static func image(
                     ^
/Carthage/Checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Snapshotting/UIView.swift:17:22: note: found this candidate
  public static func image(
                     ^
/Carthage/Checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Snapshotting/UIViewController.swift:7:13: error: ambiguous use of 'image'
    return .image()
            ^
/Carthage/Checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Snapshotting/SwiftUIView.swift:33:22: note: found this candidate
  public static func image(
                     ^
/Carthage/Checkouts/swift-snapshot-testing/Sources/SnapshotTesting/Snapshotting/UIViewController.swift:43:22: note: found this candidate
  public static func image(
                     ^

** ARCHIVE FAILED **
```

**Comment by stephencelis** [comment]
@thedavidharris Interesting...any idea how to fix or links to more info?

**Comment by zoejessica** [comment]
Just a heads up that it's breaking for me too on CircleCI builds using SPM 😞

**Comment by stephencelis** [comment]
@zoejessica 😞

Any clue as to what in Circle's environment might be affecting things? Xcode version, OS, etc.?

**Comment by zoejessica** [comment]
Sorry to dump a load of logs and run! Let's see: it was Xcode-11.4.1, running on a simulated iPhone 11 Pro on 13.5. It's building via fastlane 2.149.1

Will see if I can find something more helpful...

**Comment by thedavidharris** [comment]
It's reproducable with just a regular Product > Archive in Xcode

**Comment by srhtshn** [comment]
The PR seems to fix carthage problem. It worked here! Thanks David.

**Comment by adamwolf03** [comment]
Hi!
Can't wait to try this out.   Had some trouble setting up because I put 1.8.0 in my podfile as instructed, but it didn't work.  My first thought wasn't that 1.8.0 isn't actually available.  

If you're not able to upload 1.8.0 soon, consider updating your readme to advise that we should use 1.7.2 for now.

Thanks!
Adam

**Comment by zoejessica** [comment]
> Just a heads up that it's breaking for me too on CircleCI builds using SPM 😞

The PR also fixes the issue I was having on CircleCI

**Comment by stephencelis** [comment]
Merged and bumped to 1.8.1! Thanks @thedavidharris 😄
